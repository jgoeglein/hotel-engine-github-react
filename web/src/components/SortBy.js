import React from 'react';

export const SORT_OPTIONS = { Relevance: 'score', Stars: 'stars' };

class SortBy extends React.PureComponent {
    constructor(props) {
        super(props);
        const initialSortBy = props.sortBy || SORT_OPTIONS.Relevance;
        this.state = {
            sortBy: initialSortBy,
            sortByLabel: this.getRadioKeyFromValue(initialSortBy)
        };
        this.handleChange = this.handleChange.bind(this);
    }

    getRadioKeyFromValue(value) {
        return Reflect.ownKeys(SORT_OPTIONS).find(key => SORT_OPTIONS[key] === value);
    }

    handleChange(event) {
        this.setState({
            sortBy: event.target.value,
            sortByLabel: this.getRadioKeyFromValue(event.target.value)
        });
        if(this.props.handleChange) {
            this.props.handleChange(event.target.value);
        }
    }

    render() {
        const radios = Reflect.ownKeys(SORT_OPTIONS).map(key => 
            <label key={key}>
                <input type="radio" name="sortOptions" value={SORT_OPTIONS[key]} checked={this.state.sortByLabel === key} onChange={this.handleChange} />
                {key}
            </label>
        );
        return <div className="center-flex">
            <form>
                {radios}
            </form>
        </div>
    }
}

export default SortBy;