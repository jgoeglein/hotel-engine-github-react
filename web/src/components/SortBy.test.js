import { cleanup, fireEvent, render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import SortBy, { SORT_OPTIONS } from './SortBy';

afterEach(cleanup);

it('SortBy renders without crashing', () => {
  const tree = renderer
    .create(<SortBy />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Notifies props.handleChange correctly', () => {
    const onChange = jest.fn();
    const { getByLabelText } = render(
        <SortBy handleChange={onChange} />,
    );
    fireEvent.click(getByLabelText(/Stars/i));
    expect(onChange.mock.calls.length).toBe(1);
});

it('Renders all options', () => {
    const { queryByLabelText } = render(
        <SortBy  />,
    );
    for(const option of Reflect.ownKeys(SORT_OPTIONS)) {
        expect(queryByLabelText(option)).toBeTruthy();
    }
});