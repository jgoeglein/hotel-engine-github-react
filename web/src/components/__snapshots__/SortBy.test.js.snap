// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBy renders without crashing 1`] = `
<div
  className="center-flex"
>
  <form>
    <label>
      <input
        checked={true}
        name="sortOptions"
        onChange={[Function]}
        type="radio"
        value="score"
      />
      Relevance
    </label>
    <label>
      <input
        checked={false}
        name="sortOptions"
        onChange={[Function]}
        type="radio"
        value="stars"
      />
      Stars
    </label>
  </form>
</div>
`;
